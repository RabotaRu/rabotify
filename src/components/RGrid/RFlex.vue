<script>
  import Grid from './grid';

  export default Grid('flex');
</script>
