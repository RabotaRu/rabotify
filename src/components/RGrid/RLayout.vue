<script>
  import Grid from './grid';

  export default Grid('layout');
</script>
