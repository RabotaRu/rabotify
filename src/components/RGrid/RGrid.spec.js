import { test } from '@util/testing';
import RFlex from '@components/RGrid/RFlex.vue';
import RLayout from '@components/RGrid/RLayout.vue';

test('RFlex', ({ mount, functionalContext }) => {
  it('should conditionally apply if boolean is used', () => {
    const wrapper = mount(RFlex, functionalContext({
      attrs: {
        foo: '',
        bar: false
      }
    }));

    expect(wrapper.hasAttribute('foo')).toBe(false);
    expect(wrapper.hasAttribute('bar')).toBe(false);
    expect(wrapper.hasClass('foo')).toBe(true);
    expect(wrapper.hasClass('bar')).toBe(false);

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should pass the id attr', () => {
    const wrapper = mount(RFlex, functionalContext({
      attrs: {
        id: 'test'
      }
    }));

    expect(wrapper.find('#test')).toHaveLength(1);

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should apply different size of element', () => {
    const flex = mount(RFlex, functionalContext({
      attrs: {
        'd-flex': true,
        'xs-12': true,
        'sm-and-down-7': true
      },
      domProps: {
        innerHTML: 'Text'
      }
    }));

    const wrapper = mount(RLayout, functionalContext({
      attrs: {
        row: true,
        wrap: true
      }
    }));

    wrapper.element.appendChild(flex.element);

    expect(wrapper.html()).toMatchSnapshot();
  });
});
