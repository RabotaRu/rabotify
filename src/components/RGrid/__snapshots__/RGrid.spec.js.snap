// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RFlex should apply different size of element 1`] = `

<div class="layout row wrap">
  <div class="flex d-flex xs-12 sm-and-down-7">
    Text
  </div>
</div>

`;

exports[`RFlex should conditionally apply if boolean is used 1`] = `

<div class="flex foo">
</div>

`;

exports[`RFlex should pass the id attr 1`] = `

<div class="flex"
     id="test"
>
</div>

`;
